lib/modules/datasource/nuget/v3.ts

```diff
@@ -121,7 +122,9 @@ export async function getReleases(
   const catalogPagesQueue = catalogPages.map(
     (page) => (): Promise<CatalogEntry[]> => getCatalogEntry(http, page)
   );
-  const catalogEntries = (await p.all(catalogPagesQueue)).flat();
+  const catalogEntries = (await p.all(catalogPagesQueue))
+    .flat()
+    .sort((a, b) => versioning.sortVersions(a.version, b.version));
```

Comment:
validate versions before sort, otherwise it can throw at runtime if some wierd versions are used

Reply:
no, you need to combine with sort, otherwise we probably filter versions wrongly.

Reply:
I see the concern, especially if for some reason the NuGet feed API only returns invalid versions - but at the same time, if they aren't valid NuGet version strings we can't really reason about them here.

I think a safe fallback would be to sort all invalid versions below any valid versions, but otherwise preserve their order relative to each other as returned by the API.  Does that work for you?

Reply:
yes 🤗